### ISSUE TITLE
mini hackathon: visualization of results

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/network-meter#31]

**Comment by 0xPoly** [comment]
We discussed the possibility of using different javascript frameworks in order to enable visualization. We decided to include `d3.js` as it is one of the most popular and powerful JS visualization toolkits. We also decided to include `c3.js` as some of the simpler visualizations are painful to develop from scratch in `d3.js`.

We also discussed using `main.js` to store some developer-provided functions to parse the output.

**Comment by 0xPoly** [comment]
Alright, as agreed on a while ago, let's start with an example of what we want to achieve. This is a mockup for the 'ping' plugin. Say the user runs the equivalent of `ping -c 6 ooni.nu`, they should get as an output:

![sample](https://cloud.githubusercontent.com/assets/2194999/12517464/b764ad90-c12a-11e5-8dd0-fa58cf510b0d.png)

When clicking on on view more, the user should be shown the raw output of the plugin.

**Comment by 0xPoly** [comment]
I think the best way to implement this is to have an extra field in the plugins where developers can drop in their own code to create the visualizations. We would execute this code with `c3.js` or `d3.js` in scope, providing the output data as a variable. This would allow the developers to customize the visualizations.

I don't think this adds any extra attack surface.

**Comment by bassosimone** [comment]
I agree with what you say above and really look forward to see the ping plugin viz! :+1:
